### PR TITLE
dont keep the whole hessian

### PR DESCRIPTION
**Pull request by jelena-markovic**
changes in M_estimator not to keep the whole hessian of the loss since that's p times p matrix.

### CODE REVIEW

File: selection/randomized/M_estimator.py [hunk 1]
```diff
@@ -528,15 +528,18 @@ def new_log_density(query,
...
             logdens = 0
 
             if inactive_marginal_groups.sum()>0:
                 full_state_plus = full_state + np.multiply(limits_marginal_groups, np.array(inactive_marginal_groups, np.float))
                 full_state_minus = full_state - np.multiply(limits_marginal_groups, np.array(inactive_marginal_groups, np.float))
-                logdens += np.log(_cdf(full_state_plus) - _cdf(full_state_minus)).sum()
+                logdens += np.log(_cdf(full_state_plus) - _cdf(full_state_minus))[:,inactive_marginal_groups].sum()
 
             logdens += log_dens(full_state[:,~inactive_marginal_groups])
             return np.squeeze(logdens) # should this be negative to match the gradient log density?
```
Comment (jelena-markovic) [inline selection/randomized/M_estimator.py]: I am confused in this function. logdens here is initialized as scalar but shouldn't it be a vector of size of the number of samples (so number of rows in opt_state). When we call new_log_density internal_sample and opt_sample are matrices. Also, I assume log_dens returns a vector of the size of a given vector in its argument?

Comment (jelena-markovic) [inline selection/randomized/M_estimator.py]: I am confused by logdens, which is initialized as scalar but shouldn't it be a vector of size of the number of samples (so number of rows in opt_state). When we call new_log_density internal_sample and opt_sample are matrices.

Comment (jelena-markovic) [reply selection/randomized/M_estimator.py]: After I made this change up this function seems to run well but I am confused how.

Comment (jonathan-taylor) [reply selection/randomized/M_estimator.py]: numpy is smart enough to change the 0 later to a vector...

Comment (jelena-markovic) [reply selection/randomized/M_estimator.py]: but the LHS here
logdens += np.log(_cdf(full_state_plus) - _cdf(full_state_minus)).sum() 
is also a scalar and it should do summing row-wise, no?

**Pull request merged**